renderer/reducers/pay.js

```diff
@@ -257,20 +261,29 @@ export const queryRoutes = invoice => async (dispatch, getState) => {
           throw error
         }
 
-        // There is no guarentee that the lnd node has the Router service enabled.
+        // There is no guarantee that the lnd node has the Router service enabled.
         // Fall back to using queryRoutes if we got some other type of error.
         routes = await callQueryRoutes()
       }
     }
-
     // For older versions use queryRoutes.
     else {
       routes = await callQueryRoutes()
     }
 
-    dispatch({ type: QUERY_ROUTES_SUCCESS, routes })
+    // Check if routes match current PR to be sure we will be paying the correct PR.
+    if (isCurrentPr()) {
+      dispatch({ type: QUERY_ROUTES_SUCCESS, routes, paymentRequest })
```

Comment:
`paymentRequest` isn't used by the `QUERY_ROUTES_SUCCESS` handler so no need to pass it here